app/src/main/java/com/example/tictactoewithdatabase/BoardViewAdapter.java

```diff
@@ -0,0 +1,76 @@
+package com.example.tictactoewithdatabase;
+
+import android.content.Context;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.ImageView;
+
+import java.util.List;
+
+public class BoardViewAdapter extends BaseAdapter implements View.OnClickListener {
```

Comment:
extend ArrayAdapter<Marker> instead of BaseAdapter,
remove View.OnClickListener you don't use it